docs/how-to-guides/authentication-with-dex-gangway.md

```diff
@@ -0,0 +1,369 @@
+}
+```
+
+### Step 2: Register a New OAuth Application
+
+Go to the GitHub organization Settings
+page (`https://github.com/organizations/<your-org-name/settings/applications>`) and register a new
+OAuth application.
+
+**Set Homepage URL** to the value of the `ingress_host` field in the Dex component configuration.
+
+HomePage URL must match the `issuer_host` in Dex configuration.
+
+Set **Authorization Callback URL** to the value of the `redirect_uri` in the Dex configuration.
+
+![Registering OAuth application ](github-oauth-app-register.png?raw=true "Register a new OAuth
+applicaion")
+
+After registering the application, take note of the ClientID and ClientSecret.
+
+### Step 3: Create Variables File
+
+Create another file `lokocfg.vars` for variables and secrets that should be referenced in the cluster configuration.
+
+```
+# A random secret key (create one with `openssl rand -base64 32`)
+dex_static_client_gangway_secret="vJ09ouDw1BXEz6onT2+xW8PdofWIG8cN8+f0bv1zKZI="
+dex_static_client_gangway_id="gangway"
+
+# A random secret key (create one with `openssl rand -base64 32`)
+gangway_session_key="PMXEGiQ7fScPxuKS/DAimsCHueeWxT7HBL6I16sZzHE="
+gangway_redirect_url = "https://gangway.YOUR.CLUSTER.DOMAIN.NAME>/callback"
+
+# GitHub OAuth application client ID and secret.
+github_client_id = "87a2e79c21e7ed32re51"
+github_client_secret = "1708gg95433178e6cb63ae2f86b42b78g3810978"
+```
+
+### Step 4: Install Dex and Gangway
+
+To install, execute:
+
+```bash
+lokoctl component install
+```
+
+In few minutes cert-manager component issues the TLS certificates for the Dex and Gangway Ingress hosts.
+
+Issuing an HTTPS request to the discovery endpoint verifies the successful installation
+of Dex.
+
+```bash
+$ curl https://dex.YOUR.CLUSTER.DOMAIN.NAME/.well-known/openid-configuration
+
+{
+  "issuer": "https://dex.YOUR.CLUSTER.DOMAIN.NAME",
+  .
+  .
+  .
+}
+```
+
+To verify the Gangway installation, open the URL `https://gangway.YOUR.CLUSTER.DOMAIN.NAME` on your browser.
+
+### Step 5: Configure An API Server to Use Dex as an OIDC Authenticator
+
+Configuring an API server to use the OpenID Connect authentication plugin requires:
+
+* Deploying an API server with specific flags.
+
+* Dex is running on HTTPS.
+
+* Dex is accessible to both your browser and the Kubernetes API server.
+
+To reconfigure the API server with specific flags, edit the `kube-apiserver` DaemonSet as follows:
+
+```bash
+kubectl -n kube-system edit daemonset kube-apiserver
+```
+
+Add the following CLI arguments to the API Server container:
+
+```bash
+--oidc-issuer-url=https://dex.YOUR.CLUSTER.DOMAIN.NAME
+--oidc-client-id=gangway
+--oidc-username-claim=email
+--oidc-groups-claim=groups
+```
+
+Set the argument values according to the following table
+
+| Argument | Value |
+| ------------- | -------- |
+| `--oidc-issuer-url`| Value of the `issuer_host` in the Dex configuration. |
+| `--oidc-client-id` | The client ID obtained from GitHub in step 2. |
```

Comment:
| Argument            | Value                                                |
|---------------------|------------------------------------------------------|
| `--oidc-issuer-url` | Value of the `issuer_host` in the Dex configuration. |
| `--oidc-client-id`  | The client ID obtained from GitHub in step 2.        |
```